### PR TITLE
[MDB IGNORE] Makes only station areas, station areas.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

GLOB.the_station_areas will now only be populated with ``/area/station``:
1. ``/area/mine`` is also now exclusively the mining station
1.1. ``/area/mine/explored`` was only used once, for Tramstation maintenance, so I made it use Lavaland's explored instead, I also removed FLORA_ALLOWED from it so faunas won't spawn. This path was otherwise entirely unused and usually explored arent supposed to spawn fauna, so I thought this would be best.
1.2. ``/area/mine/unexplored`` was only used once, on Icebox's Last Ash Drake. I replaced it with Lavaland's unexplored instead, since they have their own explored/unexplored subtypes (Icemoon does too, but I didn't want to change mechanics, and this is how it was before)
2. Removes NO_ALARM flag entirely, as it was only used to exclued non-station areas from showing up in alarms, but that's not needed anymore, as we don't have non-station areas in the list anymore. I don't know if this will cause a problem on Tramstation, I'll have to see.
3. Removes shuttles and icemoon from anomaly picker, as it was no longer needed there either.

## Why It's Good For The Game

Now only the station is 'the station', which will hopefully remove many problems.

I won't have an UpdatePath here because there's a pretty important distinction between Icemoon and Lavaland subtypes, so downstreams will have to manually decide which ones they need and use those accordingly.

## Changelog

:cl:
fix: Shuttles are no longer part of the station, as they're god damn moving ships.
/:cl:

### CODE REVIEW

**Comment by msgerbs** [comment]
I don't know if this is necessarily a problem, but it is worth noting that `/area/mine/explored` allows the cult to build there but `/area/lavaland/surface` does not. So the cult won't be able to base in the area around Tram anymore.

**Comment by msgerbs** [comment]
Also if we eventually replace `is_station_level` with a check for station areas for things like revs, which probably makes sense given stuff like ice box where the station and ice moon live on the same Z, this would also make that a lot more difficult.

**Comment by JohnFulpWillard** [comment]
> I don't know if this is necessarily a problem, but it is worth noting that `/area/mine/explored` allows the cult to build there but `/area/lavaland/surface` does not. So the cult won't be able to base in the area around Tram anymore.

I saw that and thought I shouldn't add it in because I assumed it was an oversight that they were allowed there at all since it's not the station, /area/mine is just the lavaland outpost which does intentionally let cult

**Comment by msgerbs** [comment]
> > I don't know if this is necessarily a problem, but it is worth noting that `/area/mine/explored` allows the cult to build there but `/area/lavaland/surface` does not. So the cult won't be able to base in the area around Tram anymore.
> 
> I saw that and thought I shouldn't add it in because I assumed it was an oversight that they were allowed there at all since it's not the station, /area/mine is just the lavaland outpost which does intentionally let cult

I think you sent your message half-finished, but I don't see why it would be not the station. It's literally being used as part of maint.

**Comment by JohnFulpWillard** [comment]
> I think you sent your message half-finished, but I don't see why it would be not the station. It's literally being used as part of maint.

image

image


It's mostly used for outside the station, places where the Cult shouldn't be allowed to scribe. If there's some indoor areas where they can't, that should be fixed by changing their area, but Cultists shouldn't be able to draw outside of the station, so in general I think this is the proper way to do it.

**Comment by msgerbs** [comment]
Cult can normally scribe in space around the station, can't they?

**Comment by JohnFulpWillard** [comment]
> Cult can normally scribe in space around the station, can't they?

Usually there's nothing around the station to build in, since it's space. Tram is on a meteor so they're an exception
image

**Comment by msgerbs** [comment]
Huh, I thought I saw CULT_PERMITTED there. Never mind then.

File: _maps/RandomRuins/SpaceRuins/crashedship.dmm [hunk 1]
```diff
@@ -383,6 +383,7 @@
 /obj/machinery/porta_turret{
 	dir = 8;
 	installation = /obj/item/gun/energy/lasercannon;
+
```
Comment (san7890) [inline _maps/RandomRuins/SpaceRuins/crashedship.dmm:386]:
```suggestion
```

introducing some wack ass newline in the middle of the props section of a map key is a good way to instantly die i think

**Comment by thebestcodertoevercode** [comment]
Does this mean that cult can build in less areas now?

**Pull request merged**